sound/soc/sof/xtensa/core.c

```diff
@@ -132,6 +132,18 @@ static void xtensa_stack(struct snd_sof_dev *sdev, const char *level, void *oops
 				   buf, sizeof(buf), false);
 		dev_printk(level, sdev->dev, "0x%08x: %s\n", stack_ptr + i * 4, buf);
 	}
+
+	if (!xoops->plat_hdr.numaregs)
+		return;
+
+	dev_printk(level, sdev->dev, "dump ar register number %d\n", xoops->plat_hdr.numaregs);
```

Comment:
Btw, I think we should also support "cat /sys/kernel/debug/sof/exception" type of interface, so you could get the raw dump and load that gdb (-> https://docs.zephyrproject.org/latest/services/debugging/coredump.html )

Reply:
@kv2019i. I found a defect in our coredump_backend_intel_adsp_mem_window.c that we don't maintain the memory write pointer to update memory window. Zephyr coredump framework will store data into memory windows for a few times and each store will overwrite previous one.  For coredump_logging_backend_start case, the dump data is first stored in log_buf, but later send to logging system which will maintain write pointer. We need to fix this bug first then we can go on our work. Please check my patch and share your idea, thanks!

````c
void coredump_memory_dump(uintptr_t start_addr, uintptr_t end_addr)
{
 ....
	coredump_buffer_output((uint8_t *)&m, sizeof(m));

	coredump_buffer_output((uint8_t *)start_addr, len);
}
````
Adsp memory windows backend support:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/debug/coredump/coredump_backend_intel_adsp_mem_window.c#L44

````c
diff --git a/subsys/debug/coredump/coredump_backend_intel_adsp_mem_window.c b/subsys/debug/coredump/coredump_backend_intel_adsp_mem_window.c
index ddc5f48aab..df09d8a6ee 100644
--- a/subsys/debug/coredump/coredump_backend_intel_adsp_mem_window.c
+++ b/subsys/debug/coredump/coredump_backend_intel_adsp_mem_window.c
@@ -17,6 +17,7 @@
 LOG_MODULE_REGISTER(coredump_error, CONFIG_KERNEL_LOG_LEVEL);
 
 static int error;
+static int mem_ptr = 0;
 
 static void coredump_mem_window_backend_start(void)
 {
@@ -44,7 +45,7 @@ static void coredump_mem_window_backend_end(void)
 static void coredump_mem_window_backend_buffer_output(uint8_t *buf, size_t buflen)
 {
        uint32_t *mem_window_separator = (uint32_t *)(ADSP_DW->slots[1]);
-       uint8_t *mem_window_sink = (uint8_t *)(ADSP_DW->slots[1]) + 4;
+       uint8_t *mem_window_sink = (uint8_t *)(ADSP_DW->slots[1]) + 4 + mem_ptr;
        uint8_t *coredump_data = buf;
        size_t data_left;
        /* Default place for telemetry dump is in memory window. Each data is easily find using */
@@ -58,6 +59,8 @@ static void coredump_mem_window_backend_buffer_output(uint8_t *buf, size_t bufle
                        mem_window_sink++;
                        coredump_data++;
                }
+
+               mem_ptr += buflen;
        } else {
                error = -EINVAL;
        }

	} else {
		error = -EINVAL;
	}
}
````

Reply:
create a PR to fix core dump issue for memory window backend : https://github.com/zephyrproject-rtos/zephyr/pull/58465